extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/NamedAnnotationLiteral.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.core;
+
+import jakarta.enterprise.util.AnnotationLiteral;
+
+/**
+ * An interface for annotation literals that are named
+ */
+public interface NamedAnnotationLiteral {
+    AnnotationLiteral getLiteral(String name);
```

Comment:
Some kinds of beans might require that we select them through multiple qualifiers, no? To solve that possible future use case, this method should rather return an array of `AnnotationLiteral`. But because, I guess, there is no such case in sight, it is fine to keep the current solution as is and change it only once we hit such a case. I hope everybody agrees?